<a href="https://colab.research.google.com/github/MuroriM/GEM-table-to-text-through-AMR/blob/main/get_webnlg_test_human.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import json

In [ ]:
with open('/content/processed_webnlg_test.json', 'r') as f:
  entries = json.load(f)

human_annot = []
for entry in entries:
  human_annot.append((entry['text']))

with open('/content/processed_webnlg_test_human.txt', 'w') as f:
  # json.dump(human_annot, f)
  for annot in human_annot:
    f.write(annot + "\n")
print(len(human_annot))


1779


In [ ]:
with open('/content/processed_webnlg_test_human.txt', 'r') as f:
  human_annots = f.read()

print(human_annots[:3])

with open('/webnlg_gen_out_sent.txt', 'r') as f:
  machine_annots = f.read()

print(machine_annots)


print(len(human_annots))
print(len(machine_annots))

Est
Estádio is the ground of the Série 1 organization. They also play in the Brazilian Campeonato league in 2.
b>Haisheng Nie/b> (October 13, 1964) was born.
Also in Fawkham on the ground of F.C. 1 (c 0) is Fawkham.
Aleksandr is 185 m tall.
1604 leaders were in Ayala Ciudad, +.
And following the book a href="http://www.tapes-queen.com/">Tapes Quinine/a> 0 0 1969 0 0 and its prequel 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
And the population density is 1604.0, 0, above sea level 1147, and in the Pacific Ocean city of Mexico Ciudad.
9 December 1991, an epoch-era.
Death occurred in Petersburg Saint on May 27, 1703.
Population density: 1604.0 km/hr
Born Hallmann Piotr on August 25, 1987.
Code 0.0999
And the property contribution property category is located in November 2000 (the first year the establishment was established) and a distance of a mile (mile) north of the establishment.
Hebrews Modern is an officially-official language of the 1st century Jerusalem.
1. 610.0 meters above sea level

In [ ]:
import nltk
nltk.download('punkt')

from nltk.tokenize import word_tokenize

hypothesis = word_tokenize(machine_annots)
reference = word_tokenize(human_annots)
# the maximum is bigram, so assign the weight into 2 half.
BLEUscore = nltk.translate.bleu_score.sentence_bleu([reference], hypothesis, weights = (0.5, 0.5))

print(BLEUscore)

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
0.21680798024772635


In [ ]:
#Import the word_tokenize and sentence_bleu methods from NLTK.
from nltk import word_tokenize
from nltk.translate.bleu_score import sentence_bleu

#Enter the machine-translated sentence (hypothesis) and the human reference translation (reference).
hypothesis = machine_annots
reference = human_annots

#Tokenize the two sentences and store the individual tokes in two lists (one for the hypothesis tokens and one for the reference tokens).
hypothesis = word_tokenize(hypothesis)
reference = word_tokenize(reference)

#Calculate and print BLEU-1, using 1-grams as highest-order n-grams 
#Reference is placed in [square brackets] because you can score the machine-translated sentence against multiple references (applies to all BLEU score calculations listed here).
#The weights are set so that calculation is based solely on 1-gram precision.
BLEU_1 = sentence_bleu([reference], hypothesis, weights=(1, 0, 0, 0))
print(f"\nThe score for BLEU-1 is: {BLEU_1}")

#Calculate and print BLEU-2, using 2-grams as highest-order n-grams.
#The weights are set so that calculation is based on 1-gram and 2-gram precisions.
BLEU_2 = sentence_bleu([reference], hypothesis, weights=(0.5, 0.5, 0, 0))
print(f"\nThe score for BLEU-2 is: {BLEU_2}")

#Calculate and print BLEU-3, using 3-grams as highest-order n-grams.
#The weights are set so that calculation is based on 1-gram, 2-gram and 3-gram precisions.
BLEU_3 = sentence_bleu([reference], hypothesis, weights=(0.33, 0.33, 0.33, 0))
print(f"\nThe score for BLEU-3 is: {BLEU_3}")


#Calculate and print BLEU-4, using 4-grams as highest-order n-grams.
#The weights are set so that calculation is based on 1-gram, 2-gram, 3-gram and 4-gram precisions.
BLEU_4 = sentence_bleu([reference], hypothesis, weights=(0.25, 0.25, 0.25, 0.25))
print(f"\nThe score for BLEU-4 is: {BLEU_4}")

#Calculate and print BLEU-5, using 5-grams as highest-order n-grams.
#The weights are set so that calculation is based on 1-gram, 2-gram, 3-gram, 4-gram and 5-gram precisions.
BLEU_5 = sentence_bleu([reference], hypothesis, weights=(0.2, 0.2, 0.2, 0.2, 0.2))
print(f"\nThe score for BLEU-5 is: {BLEU_5}")


The score for BLEU-1 is: 0.35827305559604433

The score for BLEU-2 is: 0.21680798024772635

The score for BLEU-3 is: 0.1311809159463567

The score for BLEU-4 is: 0.07874044322820561

The score for BLEU-5 is: 0.04903498385061269


In [ ]:
import nltk
nltk.download('punkt')

from nltk.tokenize import word_tokenize

totalBleu = 0
for i in range(len(machine_annots)):

  hypothesis = word_tokenize(machine_annots[i])
  reference = word_tokenize(human_annots[i])
  # the maximum is bigram, so assign the weight into 2 half.
  BLEUscore = nltk.translate.bleu_score.sentence_bleu([reference], hypothesis, weights = (0.5, 0.5))
  # print(BLEUscore)
  totalBleu += BLEUscore

print(totalBleu / len(machine_annots))

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!


/usr/local/lib/python3.7/dist-packages/nltk/translate/bleu_score.py:490: UserWarning: 
Corpus/Sentence contains 0 counts of 2-gram overlaps.
BLEU scores might be undesirable; use SmoothingFunction().
  warnings.warn(_msg)


0.1558387678121714



[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
/usr/local/lib/python3.7/dist-packages/nltk/translate/bleu_score.py:490: UserWarning: 
Corpus/Sentence contains 0 counts of 2-gram overlaps.
BLEU scores might be undesirable; use SmoothingFunction().
  warnings.warn(_msg)
0.1558387678121714